### ISSUE TITLE
Exception in SerializationDumper.java - line.1514 - throws unexpected identifier for objectd field value 0x00

### ISSUE THREAD

**Issue by johnhines13** [NickstaDB/SerializationDumper#5]
```
java -jar SerializationDumper.jar -r  byte-stream-InnerString > InnerString.decode

SerializationDumper.java:  exception context:
1513                         default:                                //Unknown/unsupported
1514                                 throw new RuntimeException("Error: Unexpected identifier for obje     ct field value 0x" + this.byteToHex(this._data.peek()));

Exception in thread "main" java.lang.RuntimeException: Error: Unexpected identifier for object field value 0x00
        at nb.deser.SerializationDumper.readObjectField(SerializationDumper.java:1514)
        at nb.deser.SerializationDumper.readFieldValue(SerializationDumper.java:971)
        at nb.deser.SerializationDumper.readClassDataField(SerializationDumper.java:921)
        at nb.deser.SerializationDumper.readClassData(SerializationDumper.java:859)
        at nb.deser.SerializationDumper.readNewObject(SerializationDumper.java:455)
        at nb.deser.SerializationDumper.readContentElement(SerializationDumper.java:347)
        at nb.deser.SerializationDumper.parseStream(SerializationDumper.java:319)
        at nb.deser.SerializationDumper.main(SerializationDumper.java:101)
```

From Test program:   Create some objects and set some values.
```java
Employee emp = new Employee();
Address addr = new Address();
ComplexObject co = new ComplexObject("a","b");
InnerString innerstring = new InnerString("x","y");
addr.setStreet("101 Maple Avenue");
addr.setCity("San Mateo");
addr.setZipcode(95616);
emp.setAddress(addr);
emp.setName("John Hines");
 emp.setAge(22);

// Prepare stream by stuffing objects in it  -- process just InnerString
Object obj;  
 objectOutputStream = new 
      ObjectOutputStream(byteArrayOutputStream);
// objectOutputStream.writeObject(addr);   // Address - Decodes OKAY
// objectOutputStream.writeObject(emp);   // Employee - Decodes OKAY
**objectOutputStream.writeObject(innerstring);  // InnerString - Decodes with Exception**
// objectOutputStream.writeObject(co);   // ComplexObject - Decodes with Excepton
objectOutputStream.close();

// Pull the bytes out of the objectOutputStream
bytes = byteArrayOutputStream.toByteArray();
// Store bytes to ./byte-stream
**String FILEPATH = "./byte-stream-InnerString";** 
File file = new File(FILEPATH);          
OutputStream os = new FileOutputStream(file); 
os.write(bytes); 
os.close(); 
```

```
$ hexdump -C byte-stream-InnerString 
00000000  ac ed 00 05 73 72 00 12  62 75 67 74 61 67 2e 49  |....sr..bugtag.I|
00000010  6e 6e 65 72 53 74 72 69  6e 67 da e0 f6 fb ad 97  |nnerString......|
00000020  62 8d 02 00 03 4c 00 07  64 6f 72 6b 69 73 68 74  |b....L..dorkisht|
00000030  00 12 4c 62 75 67 74 61  67 2f 53 69 6e 67 6c 65  |..Lbugtag/Single|
00000040  53 74 72 3b 4c 00 09 69  6e 6e 65 72 43 68 61 72  |Str;L..innerChar|
00000050  74 00 12 4c 62 75 67 74  61 67 2f 49 6e 6e 65 72  |t..Lbugtag/Inner|
00000060  43 68 61 72 3b 4c 00 09  69 6e 6e 65 72 53 74 72  |Char;L..innerStr|
00000070  41 74 00 12 4c 6a 61 76  61 2f 6c 61 6e 67 2f 53  |At..Ljava/lang/S|
00000080  74 72 69 6e 67 3b 78 70  73 72 00 10 62 75 67 74  |tring;xpsr..bugt|
00000090  61 67 2e 53 69 6e 67 6c  65 53 74 72 fa 8d d0 44  |ag.SingleStr...D|
000000a0  7c d1 ce 5f 02 00 03 4c  00 01 64 71 00 7e 00 03  ||.._...L..dq.~..|
000000b0  4c 00 04 64 6f 72 6b 71  00 7e 00 03 4c 00 01 65  |L..dorkq.~..L..e|
000000c0  71 00 7e 00 03 78 70 74  00 04 5a 31 30 35 74 00  |q.~..xpt..Z105t.|
000000d0  03 5a 31 30 74 00 0c 53  6d 69 6c 65 79 20 3a 2d  |.Z10t..Smiley :-|
000000e0  29 31 31 73 72 00 10 62  75 67 74 61 67 2e 49 6e  |)11sr..bugtag.In|
000000f0  6e 65 72 43 68 61 72 0b  ad b6 d5 dd 81 90 73 02  |nerChar.......s.|
00000100  00 02 43 00 06 69 6e 6e  65 72 43 43 00 06 69 6e  |..C..innerCC..in|
00000110  6e 65 72 44 78 70 00 53  00 54 74 00 02 78 79     |nerDxp.S.Tt..xy|
0000011f
```

Test Object: InnerString
```
$ cat -n InnerString.java
     1  package bugtag;
     2  import java.io.ByteArrayInputStream;
     3  import java.io.ByteArrayOutputStream;
     4  import java.io.DataOutputStream;
     5  import java.io.IOException;
     6  import java.io.ObjectInputStream;
     7  import java.io.ObjectOutputStream;
     8  import java.io.ObjectStreamConstants;
     9  import java.io.Serializable;
    10  import java.util.ArrayList;
    11  import java.util.HashMap;
    12  import java.util.List;
    13  import java.util.Map;
    14
    15  public class InnerString implements Serializable {
    16     String innerStrA = null;
    17     InnerChar innerChar = new InnerChar('S', 'T');
    18     SingleStr dorkish = new SingleStr("Smiley :-)");
    19     public InnerString(String a, String b) {
    20         this.innerStrA = a + b;
    21     }
    22  }
```

Test Object: SingleStr
```
[jhines@gps-dev inputstreamfw]$ cat -n SingleStr.java
     1  package bugtag;
     2  import java.io.ByteArrayInputStream;
     3  import java.io.ByteArrayOutputStream;
     4  import java.io.DataOutputStream;
     5  import java.io.IOException;
     6  import java.io.ObjectInputStream;
     7  import java.io.ObjectOutputStream;
     8  import java.io.ObjectStreamConstants;
     9  import java.io.Serializable;
    10  import java.util.ArrayList;
    11  import java.util.HashMap;
    12  import java.util.List;
    13  import java.util.Map;
    14
    15  public class SingleStr implements Serializable { 
    16     String dork, d, e;
    17     SingleStr(String dork) { 
    18         this.dork = "Z" + dork.length(); 
    19         this.d=this.dork+5; 
    20         this.e=dork+11;
    21     };
    22  }
```

Attached are byte-stream-ComplexObject and byte-stream-InnerString

**Comment by johnhines13** [comment]
base64 < byte-stream-InnerString 
rO0ABXNyABJidWd0YWcuSW5uZXJTdHJpbmfa4Pb7rZdijQIAA0wAB2Rvcmtpc2h0ABJMYnVndGFnL1NpbmdsZVN0cjtMAAlpbm5lckNoYXJ0ABJMYnVndGFnL0lubmVyQ2hhcjtMAAlpbm5lclN0ckF0
ABJMamF2YS9sYW5nL1N0cmluZzt4cHNyABBidWd0YWcuU2luZ2xlU3Ry+o3QRHzRzl8CAANMAAFkcQB+AANMAARkb3JrcQB+AANMAAFlcQB+AAN4cHQABFoxMDV0AANaMTB0AAxTbWlsZXkgOi0pMTFzcgAQYnVndGFnLklubmVyQ2hhcgutttXdgZBzAgACQwAGaW5uZXJDQwAGaW5uZXJEeHAUwBUdAACeHk=

**Comment by johnhines13** [comment]
base64 < byte-stream-ComplexObject 
rO0ABXNyABRidWd0YWcuQ29tcGxleE9iamVjdHsKfByaU0swAgAMQwAGY1ZhbHVlQwAJY1ZhbHVlQmlnSQAIaW50VmFsdWVMAAhkb3JrYWJsZXQAFkxqYXZhL2lvL1NlcmlhbGl6YWJsZTtMAAZpbm5lcjF0ABJMYnVndGFnL0lubmVyQ2hhcjtMAAdpbm5lclMxdAAUTGJ1Z3RhZy9Jbm5lclN0cmluZztMAARsaXN0dAAQTGphdmEvdXRpbC9MaXN0O0wAA21hcHQAD0xqYXZhL3V0aWwvTWFwO1sAB3N0cmluZ3N0ABNbTGphdmEvbGFuZy9TdHJpbmc7TAAHeFN0cmluZ3QAEkxqYXZhL2xhbmcvU3RyaW5nO0wAB3lTdHJpbmdxAH4AB0wAB3pTdHJpbmdxAH4AB3hwAEMBAQAAAP5zcgAQYnVndGFnLlNpbmdsZVN0cvqN0ER80c5fAgADTAABZHEAfgAHTAAEZG9ya3EAfgAHTAABZXEAfgAHeHB0AANaNTV0AAJaNXQAB0RPUksyMTFzcgAQYnVndGFnLklubmVyQ2hhcgutttXdgZBzAgACQwAGaW5uZXJDQwAGaW5uZXJEeHAARwBIc3IAEmJ1Z3RhZy5Jbm5lclN0cmluZ9rg9vutl2KNAgADTAAHZG9ya2lzaHQAEkxidWd0YWcvU2luZ2xlU3RyO0wACWlubmVyQ2hhcnEAfgACTAAJaW5uZXJTdHJBcQB+AAd4cHNxAH4ACXQABFoxMDV0AANaMTB0AAxTbWlsZXkgOi0pMTFzcQB+AA4AUwBUdAAOU3RyaW5nQVN0cmluZ0JzcgATamF2YS51dGlsLkFycmF5TGlzdHiB0h2Zx2GdAwABSQAEc2l6ZXhwAAAAA3cEAAAAA3QADkRvb2xleSBDb21wbGV4dAASRG9vbGV5IENvbXBsZWN0aW9udAARRG9vbGV5IFJFRkxFQ1RJT054c3IAEWphdmEudXRpbC5IYXNoTWFwBQfawcMWYNEDAAJGAApsb2FkRmFjdG9ySQAJdGhyZXNob2xkeHA/QAAAAAAADHcIAAAAEAAAAAJ0AAFhc3IAEWphdmEubGFuZy5JbnRlZ2VyEuKgpPeBhzgCAAFJAAV2YWx1ZXhyABBqYXZhLmxhbmcuTnVtYmVyhqyVHQuU4IsCAAB4cAAAAQF0AAFic3EAfgAhAAAEAXh1cgATW0xqYXZhLmxhbmcuU3RyaW5nO63SVufpHXtHAgAAeHAAAAAGdAAIU3RyaW5nQTB0AAhTdHJpbmdBMXQACFN0cmluZ0EycHQAB1N0cmluZ0xwdAAJRWtzU3RyaW5ncQB+ACBxAH4AJA==

**Comment by johnhines13** [comment]
"java -jar SerializationDumper.jar -r byte-stream-Filename"  is what I use to decode the raw byte stream created by my  test program above.   You can take the 2 base-64 encodings and decode them into a binary file.   Then use -r option of SerializationDumper to attempt to decode.  These two throw exceptions.  I have other test classes  that do not throw exceptions.  The utility does a very good job on other objects that I am testing with.

**Comment by NickstaDB** [comment]
Thanks @johnhines13, this is fixed in commit 602c75fc17ddc611119cc11a50fa4c4032c016a8.

The code was incorrectly reading `char` field values as one byte when they're actually two bytes.
